src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
I think this will be a really nice utility in a library. 

1. Does exist one already?
2. 1 is negative, we can create one under @verdaccio/ scope.